CI: best-effort Playwright font install leaks an apt lock that fails the fatal Redis install in playwright-mock.yml

## What happens

`MCP list_changed (replica count 1)` and `(replica count 2)` in `.github/workflows/playwright-mock.yml` fail during environment setup, before any test body runs:

```
E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 3253 (apt-get)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?
##[error]Process completed with exit code 100.
```

The job then uploads no report (`No files were found with the provided path: e2e/playwright-report/**`), so the failure looks like a test failure but nothing was executed.

## Root cause

`.github/scripts/install-playwright-fonts.sh` runs `timeout 70 npx playwright install-deps chrome`. When that timeout fires, it kills `npx`, but the `sudo apt-get` it spawned survives and keeps holding `/var/lib/dpkg/lock-frontend`. The script's remaining attempts fail on that lock and it exits `0` by design, because fonts are best effort — but the orphaned `apt-get` is still holding the lock when the script returns.

The next step, **"Install Redis runtime dependencies"**, is deliberately fatal:

```yaml
      # Redis is a hard requirement for this job, so this step stays fatal.
      - name: Install Redis runtime dependencies
        timeout-minutes: 5
        run: |
          sudo apt-get update
          sudo apt-get install -y redis-server redis-tools
```

It hits the same lock and exits 100, so Redis never installs and the job dies. A best-effort step is taking down a fatal one.

## Why it is intermittent

It only fires when the Ubuntu mirror is slow enough for the 70s per-attempt cap to trip. A failing run logged `Fetched 11.3 MB in 44s (255 kB/s)`. On faster runs `install-deps` completes and nothing leaks, which is why the same workflow passes on many runs.

## Suggested fix

Wait for the dpkg lock to clear before the fatal step:

```yaml
      - name: Install Redis runtime dependencies
        timeout-minutes: 5
        run: |
          # The best-effort font step can leave an orphaned apt-get holding the
          # dpkg lock after its `timeout` fires; wait it out rather than failing.
          for _ in $(seq 1 60); do
            sudo fuser /var/lib/dpkg/lock-frontend >/dev/null 2>&1 || break
            sleep 2
          done
          sudo apt-get update
          sudo apt-get install -y redis-server redis-tools
```

Having `install-playwright-fonts.sh` wait for the lock to clear before it exits would fix it at the source instead, so no later apt consumer inherits the problem.

`.github/workflows/cache-integration-tests.yml` (around line 50) installs `redis-server redis-tools` the same way and would be exposed to the same race if a best-effort apt step were ever added before it.

--- a/.github/workflows/playwright-mock.yml
+++ b/.github/workflows/playwright-mock.yml
@@ -251,12 +251,17 @@ jobs:
         continue-on-error: true
         run: timeout -k 10 90 npx playwright install ffmpeg
 
+      # This job deliberately skips the optional font install: its bounded
+      # Playwright apt process can outlive the wrapper on a slow mirror and
+      # retain the package-manager lock needed by the required Redis install.
+      # The MCP suite does not enable visual snapshot assertions.
+
       # Redis is a hard requirement for this job, so this step stays fatal.
       - name: Install Redis runtime dependencies
         timeout-minutes: 5
         run: |
-          sudo apt-get update
-          sudo apt-get install -y redis-server redis-tools
+          sudo apt-get -o DPkg::Lock::Timeout=300 update
+          sudo apt-get -o DPkg::Lock::Timeout=300 install -y redis-server redis-tools
 
       - name: Start standalone Redis and Redis Cluster
         run: |

